numpy/core/overrides.py

```diff
@@ -68,24 +68,26 @@
 
 def verify_matching_signatures(implementation, dispatcher):
     """Verify that a dispatcher function has the right signature."""
-    # Get an ordered dict of Parameter objects from each signature
-    implementation_sig = inspect.signature(implementation).parameters
-    dispatcher_sig = inspect.signature(dispatcher).parameters
-
-    for p1, p2 in zip(implementation_sig.values(), dispatcher_sig.values()):
-        if (
-            (p1.name != p2.name) or (p1.kind != p2.kind) or
-            ((p1.default is p1.empty) and (p2.default is not p2.empty))
-        ):
-            raise RuntimeError(
-                f'implementation and dispatcher for {implementation} have '
-                 'different function signatures'
-            )
-        if (p1.default is not p1.empty) and (p2.default is not None):
-            raise RuntimeError(
-                f'dispatcher functions can only use None for default '
-                 'argument values'
+    implementation_sig = inspect.signature(implementation)
+    dispatcher_sig = inspect.signature(dispatcher)
+
+    implementation_sig_with_none_defaults = implementation_sig.replace(
+        parameters=[
+            p.replace(
+                annotation=p.empty,
```

Comment:
Using `p.empty` seems weird here since `.empty` is static, and the docs for inspect suggest the static spelling.